### PR TITLE
[compaction]Add local sample config for sort compaction

### PR DESCRIPTION
**Pull request by wg1026688210**
### Purpose

Add local sample config for sort compaction . It' helpful for the sort compaction task with large parallelism,which sampling may lead to a performance bottleneck.

### CODE REVIEW

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/sorter/SortUtils.java [hunk 1]
```diff
@@ -99,8 +99,13 @@ public static <KEY> DataStream<RowData> sortStreamByKey(
                     "The adaptive batch scheduler is not supported. Please set the sink parallelism using the key: "
                             + FlinkConnectorOptions.SINK_PARALLELISM.key());
         }
-        final int sampleSize = sinkParallelism * 1000;
-        final int rangeNum = sinkParallelism * 10;
+        final int localSampleSize =
+                options.getLocalSampleSize().orElseGet(() -> sinkParallelism * 1000);
+        final int globalSampleSize =
+                options.getGlobalSampleSize().orElseGet(() -> sinkParallelism * 1000);
+        final int rangeNum = options.getSortRange().orElseGet(() -> sinkParallelism * 10);
```
Comment (leaves12138) [inline paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/sorter/SortUtils.java:106]: I think there is no need to make localSampleSize and globalSampleSize different, if you want a performance improvement when parallelism is large, just add one parameter of sampleSize option. 20 or 2000 times, default 1000. rangeNum still sinkParallelism * 10.

**Review by leaves12138 — CHANGES_REQUESTED** [review]
Thanks for @wg1026688210, left some comment

**Comment by wg1026688210** [comment]
@leaves12138  cc

File: paimon-common/src/main/java/org/apache/paimon/CoreOptions.java [hunk 2]
```diff
@@ -1056,6 +1056,12 @@ public class CoreOptions implements Serializable {
                                     + "If the data size allocated for the sorting task is uneven,which may lead to performance bottlenecks, "
                                     + "the config can be set to size.");
 
+    public static final ConfigOption<Integer> SORT_COMPACTION_LOCAL_SAMPLE_SIZE =
+            key("sort-compaction.local-sample.size")
+                    .intType()
+                    .noDefaultValue()
+                    .withDescription(
+                            "The size of local sample for sort-compaction.By default,the value is the sink parallelism * 1000.");
```
Comment (leaves12138) [inline paimon-common/src/main/java/org/apache/paimon/CoreOptions.java:1064]: I think the parameter could be magnification.  int localSampleSize = parallelism * magnification.   magnification should be greater than or equal to 20

**Review by leaves12138 — CHANGES_REQUESTED** [review]
Thanks for @wg1026688210 , left one comment.

File: paimon-common/src/main/java/org/apache/paimon/CoreOptions.java [hunk 3]
```diff
@@ -1056,6 +1056,12 @@ public class CoreOptions implements Serializable {
                                     + "If the data size allocated for the sorting task is uneven,which may lead to performance bottlenecks, "
                                     + "the config can be set to size.");
 
+    public static final ConfigOption<Integer> SORT_COMPACTION_LOCAL_SAMPLE_SIZE =
```
Comment (leaves12138) [inline paimon-common/src/main/java/org/apache/paimon/CoreOptions.java:1059]: SORT_COMPACTION_SAMPLE_MAGNIFICATION

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/sorter/SortUtils.java [hunk 4]
```diff
@@ -99,9 +99,18 @@ public static <KEY> DataStream<RowData> sortStreamByKey(
                     "The adaptive batch scheduler is not supported. Please set the sink parallelism using the key: "
                             + FlinkConnectorOptions.SINK_PARALLELISM.key());
         }
-        final int sampleSize = sinkParallelism * 1000;
+        final int localSampleSize =
+                options.getLocalSampleSize().orElseGet(() -> sinkParallelism * 1000);
+        final int globalSampleSize = sinkParallelism * 1000;
         final int rangeNum = sinkParallelism * 10;
 
+        if (localSampleSize < 20) {
```
Comment (leaves12138) [inline paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/sorter/SortUtils.java:107]: if(localSampleSignification < 20) throws

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/sorter/SortUtils.java [hunk 5]
```diff
@@ -99,9 +99,18 @@ public static <KEY> DataStream<RowData> sortStreamByKey(
                     "The adaptive batch scheduler is not supported. Please set the sink parallelism using the key: "
                             + FlinkConnectorOptions.SINK_PARALLELISM.key());
         }
-        final int sampleSize = sinkParallelism * 1000;
+        final int localSampleSize =
```
Comment (leaves12138) [inline paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/sorter/SortUtils.java:102]: localSampleSize = sinkParallelism * sampleSignification

Comment (wg1026688210) [reply paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/sorter/SortUtils.java]: done~

**Review by leaves12138 — APPROVED** [review]
+1

**Pull request merged**